library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitleViewUtils.java

```diff
@@ -48,5 +54,27 @@ public static float resolveTextSize(
     }
   }
 
+  public static void preserveJapaneseLanguageFeatures(Spannable copy, Spanned original) {
```

Comment:
Instead of needing to encode this list of Span types in SubtitleViewUtils, what do you think about using a marker interface like `LanguageFeature` that would be implemented by spans like `RubySpan`?

Reply:
Great idea! Will update the PR